github-upload-action-secrets

```diff
@@ -46,10 +48,51 @@ def encrypt(public_key: str, secret_value: str) -> str:
 
     public_key = public.PublicKey(public_key.encode("utf-8"), encoding.Base64Encoder())
     sealed_box = public.SealedBox(public_key)
-    encrypted = sealed_box.encrypt(secret_value.encode("utf-8"))
+    encrypted = sealed_box.encrypt(secret_value)
     return b64encode(encrypted).decode("utf-8")
 
 
+def generate_ssh_key(pubkey):
+    """Generates a Ed25519 keypair and outputs public/private in OpenSSH format"""
+    """We could also do this, using cryptography.hazmat.primitives:
+
+    key = Ed25519PrivateKey.generate()
+    public = key.public_key().public_bytes(Encoding.OpenSSH, PublicFormat.OpenSSH)
+    private = key.private_bytes(Encoding.PEM, PrivateFormat.OpenSSH, NoEncryption())
+
+    but we make a conscious design decision to use ssh-keygen, regardless of
+    how awkward it is: ssh-keygen knows how to make good SSH keys.  The word
+    "hazmat" in the module name is enough of a warning here about the dangers
+    of getting this stuff wrong.
+
+    See discussion in https://github.com/cockpit-project/bots/pull/2192
+    """
+    with tempfile.TemporaryDirectory() as dir:
+        key = f'{dir}/link-to-fd3-and-fd4'
+
+        os.symlink('/proc/self/fd/3', f'{key}')
```

Comment:
This is clearly overdoing f-strings :grin:

Reply:
So, obviously, visual uniformity with the line below.

I thought about

```
  , key)
  , key + '.pub')
```

but you know me: I'm a sucker for f-strings :)